simplified-app-shared/src/main/java/org/nypl/simplified/app/AnnotationsManager.kt

```diff
@@ -0,0 +1,600 @@
+                                        completion: (enableSync: Boolean) -> Unit)
+  {
+    if (!syncIsPossible(account)) {
+      LOG.debug("Account does not satisfy conditions for sync setting request.")
+      completion(false)
+      return
+    }
+
+    //TODO I think this condition is either wrong or out of date.. think about it some more during testing
+    if (Simplified.getSharedPrefs().getBoolean("userHasSeenFirstTimeSyncMessage") == true &&
+        Simplified.getSharedPrefs().getBoolean("syncPermissionGranted", libraryAccount.id) == false) {
+      completion(false)
+      return
+    }
+
+    syncPermissionStatusUriRequest { initialized, syncIsPermitted ->
+      if (initialized && syncIsPermitted) {
+        Simplified.getSharedPrefs().putBoolean("userHasSeenFirstTimeSyncMessage", true)
+        LOG.debug("Sync has already been enabled on the server. Enable here as well.")
+        completion(true)
+      } else if (!initialized && !Simplified.getSharedPrefs().getBoolean("userHasSeenFirstTimeSyncMessage")) {
+        LOG.debug("Sync has never been initialized for the patron. Proceeding with opt-in alert.")
+        presentFirstTimeSyncAlertDialog(completion)
+      } else {
+        LOG.debug("Sync has been initialized and not permitted. Continuing with sync disabled.")
+        completion(false)
+      }
+    }
+  }
+
+  private fun syncPermissionStatusUriRequest(completion: (initialized: Boolean, syncIsPermitted: Boolean) -> Unit)
+  {
+    val catalogUriString = libraryAccount.catalogUrl ?: null
+    val baseUri = if (catalogUriString != null) Uri.parse(catalogUriString) else null
+    val permissionRequestUri = if (baseUri != null) Uri.withAppendedPath(baseUri, "patrons/me/") else null
+
+    if (permissionRequestUri == null) {
+      LOG.debug("Could not create Annotations URL from Main Feed URL. Abandoning attempt to retrieve sync setting.")
+      return
+    }
+
+    val request = NYPLJsonObjectRequest(
+        Request.Method.GET,
+        permissionRequestUri.toString(),
+        credentials.barcode.toString(),
+        credentials.pin.toString(),
+        null,
+        Listener<JSONObject> { response ->
+          handleSyncPermission(response, completion)
+        },
+        Response.ErrorListener { error ->
+          LOG.error("GET request fail! Error: ${error.message}")
+        })
+
+    request.retryPolicy = DefaultRetryPolicy(
+        60 * 1000,
+        DefaultRetryPolicy.DEFAULT_MAX_RETRIES,
+        DefaultRetryPolicy.DEFAULT_BACKOFF_MULT)
+
+    requestQueue.add(request)
+  }
+
+  private fun handleSyncPermission(response: JSONObject,
+                                   completion: (initialized: Boolean, syncIsPermitted: Boolean) -> Unit)
+  {
+    try {
+      val settings = response.getJSONObject("settings")
+      val syncSettingsBool = settings.getBoolean("simplified:synchronize_annotations")
+      completion(true, syncSettingsBool)
```

Comment:
not to open up a can of worms, but are we guaranteed that this is called on a UI Thread? do we need such guarantees?

Reply:
Not gauranteed, and I don't imagine getting there very easily, but it can't hurt to check and throw an exception for done the road. `UIThread.checkIsUIThread()` is used in this way in many other spots of the app 👍